src/compiler/commandLineParser.ts

```diff
@@ -2023,9 +2023,8 @@ namespace ts {
         };
     }
 
-    function validateSpecs(specs: string[], errors: Diagnostic[], allowTrailingRecursion: boolean, jsonSourceFile: JsonSourceFile, specKey: string) {
-        const validSpecs: string[] = [];
-        for (const spec of specs) {
+    function validateSpecs(specs: string[], errors: Push<Diagnostic>, allowTrailingRecursion: boolean, jsonSourceFile: JsonSourceFile, specKey: string) {
+        return specs.filter(spec => {

```

Comment:
I'm not sure this is an improvement in readability what with all the side effects elsewhere in the block. Filters usually have small predicates.